src/olympia/blocklist/tests/test_cron.py

```diff
@@ -538,49 +525,215 @@ def test_upload_stash_unless_enough_changes(self):
+            'softblocked': 0,
+            # There are no unblocked versions
+            'unblocked': 0,
+        }
+
+        self._test_upload_stash_and_filter(
+            # Even though there are enough soft blocks, soft blocking is disabled
+            enable_soft_blocking=False,
+            expected_stash=expected_stash,
+            # Expect no filter as soft blocking is disabled
+            expected_filters=[],
+        )
+
+        # Now try again with soft blocking enabled
+        self._test_upload_stash_and_filter(
+            # Soft blocking is enabled, so we expect the same stash and a new filter
+            enable_soft_blocking=True,
+            expected_stash=expected_stash,
+            # Expect a soft blocked filter
+            expected_filters=[BlockType.SOFT_BLOCKED],
+        )
+
+    def test_upload_blocked_stash_and_softblock_filter_with_switch_enabled(self):
+        # Enough blocks for a stash
         self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
+        # Enough soft blocks for a filter
+        self._block_version(is_signed=True, block_type=BlockType.SOFT_BLOCKED)
         self._block_version(is_signed=True, block_type=BlockType.SOFT_BLOCKED)
-        upload_mlbf_to_remote_settings()
-        assert self.mocks[
-            'olympia.blocklist.cron.upload_filter.delay'
-        ].call_args_list == [
-            mock.call(
-                self.current_time,
-                filter_list=[BlockType.BLOCKED.name],
-                create_stash=False,
-            )
-        ]
-        mlbf = MLBF.load_from_storage(self.current_time)
-        assert mlbf.storage.exists(mlbf.filter_path(BlockType.BLOCKED))
-        assert not mlbf.storage.exists(mlbf.stash_path)
 
-        with override_switch('enable-soft-blocking', active=True):
-            self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
-            self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
-            upload_mlbf_to_remote_settings()
-            self.mocks['olympia.blocklist.cron.upload_filter.delay'].assert_called_with(
-                self.current_time,
-                filter_list=[BlockType.BLOCKED.name],
-                create_stash=False,
-            )
-            mlbf = MLBF.load_from_storage(self.current_time)
-            assert mlbf.storage.exists(mlbf.filter_path(BlockType.BLOCKED))
-            assert not mlbf.storage.exists(mlbf.stash_path)
+        self._test_upload_stash_and_filter(
+            # Soft blocking is enabled, so we expect a stash and a filter
+            enable_soft_blocking=True,
+            expected_stash={
+                'blocked': 1,
+                # Expect no soft blocks becuase there are enough for a filter
+                'softblocked': 0,
+                'unblocked': 0,
+            },
+            # Expect a soft blocked filter
+            expected_filters=[BlockType.SOFT_BLOCKED],
+        )
```

Comment:
Aren't these two tests the same? I don't think we need `test_upload_blocked_stash_and_softblock_filter_with_switch_enabled`, it is covered by the second call to `_test_upload_stash_and_filter` in `test_upload_blocked_stash_and_softblock_filter`, no?